worked-solutions/01_penguin_classification_solutions.ipynb

```diff
@@ -520,17 +563,20 @@
     "\n",
     "- Before we jump in and write these loops, we must first choose an activation function to apply to the model's outputs.\n",
     "  - Here we are going to use the softmax activation function: see [the PyTorch docs](https://pytorch.org/docs/stable/generated/torch.nn.Softmax.html).\n",
-    "  - For those of you who've studied physics, you may be remininded of the partition function in thermodynamics.\n",
     "  - This activation function is good for classifcation when the result is one of ``A or B or C``.\n",
     "    - It's bad if you even want to assign two classification to one images—say a photo of a dog _and_ a cat.\n",
     "  - It turns the raw outputs, or logits, into \"psuedo probabilities\", and we take our prediction to be the most probable class.\n",
     "\n",
+    "\n",
+    "- The purpose of training loop is that for each epoch, all the mini-batches are sequentially passed through the model to make predictions, compute the loss, and update the model parameters and repeat it until we receive a satisfactory performance.\n",
+    "- The validation loop works similarly, but the purpose is to validate the model training's performance without any weight updates. Hence, we are not calculating any gradient during validation. This is made sure using the ``no_grad`` decorator with validation loop.\n",
```

Comment:
The `@no_grad` decorator disables gradient tracking in the function. We could point out that it is optional. 

We also don't need gradients in the batch_level_accuracy() function.